### PR TITLE
Fix arm64 mac symbols generation

### PR DESCRIPTION
**Pull request by atuchin-m**
Resolves https://github.com/brave/brave-browser/issues/23696

The main problem: arm64 `dump_syms` was used on intel mac.
Fixed the same way as chromium does https://github.com/chromium/chromium/blob/main/chrome/BUILD.gn#L1298

Also Chromium already have scripts to generate breakpad symbols and we call it (yep, double symbols generation).
I've created [an issue](https://github.com/brave/brave-browser/issues/24125) to reuse them, brave scripts like `generate_breakpad_symbols.py` should be retired.

P.S. I have to format/change some lines to make `pylint` happy.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: tools/posix/generate_breakpad_symbols.py [hunk 1]
```diff
@@ -198,20 +188,24 @@ def _Worker():
...
                     binary = GetSymbolPath(options, binary)
 
-                syms = GetCommandOutput([GetDumpSymsBinary(options.build_dir), '-r', '-c', binary])
-                module_line = re.match("MODULE [^ ]+ [^ ]+ ([0-9A-F]+) (.*)\n", syms)
-                output_path = os.path.join(options.symbols_dir, module_line.group(2),
+                dump_syms_bin = options.dump_syms_bin
+                if not os.access(dump_syms_bin, os.X_OK):
+                    raise Exception(f'Cannot find dump_syms: {dump_syms_bin}')
+
+                syms = GetCommandOutput([dump_syms_bin, '-c', binary])
```
Comment (atuchin-m) [inline tools/posix/generate_breakpad_symbols.py:195]: `-r` is removed (` Do not handle inter-compilation unit references`).
With the flag there are a lot of errors and the resulting symbols seems to be nearly empty.

**Comment by atuchin-m** [comment]
test-brave-browser-build-macos-arm64 build (in-progress):
https://ci.brave.com/blue/organizations/jenkins/test-brave-browser-build-macos-arm64/detail/test-brave-browser-build-macos-arm64/207/pipeline

**Review by goodov — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

**Comment by mkarolin** [comment]
@atuchin-m I think this may have broken Linux dist builds: 
https://ci.brave.com/view/linux/job/test-brave-browser-build-linux-x64/648/execution/node/237/log/

<pre>
 FAILED: dist/brave.breakpad.syms 
 python3 ../../brave/tools/posix/generate_breakpad_symbols.py --symbols-dir=/home/ubuntu/workspace/test-brave-browser-build-linux-x64-nightly/src/out/Release/dist/brave.breakpad.syms --jobs=16 --build-dir=/home/ubuntu/workspace/test-brave-browser-build-linux-x64-nightly/src/out/Release --binary=\[\"brave\"\] --libchromiumcontent-dir=/home/ubuntu/workspace/test-brave-browser-build-linux-x64-nightly/src/ --clear
 usage: generate_breakpad_symbols.py [-h] --build-dir BUILD_DIR --symbols-dir
                                     SYMBOLS_DIR --libchromiumcontent-dir
                                     LIBCHROMIUMCONTENT_DIR --binary BINARY
                                     --dump-syms-bin DUMP_SYMS_BIN [--clear]
                                     [-j JOBS] [-v]
 generate_breakpad_symbols.py: error: the following arguments are required: --dump-syms-bin
</pre>

**Comment by atuchin-m** [comment]
Thanks, I missed this. I will it soon

**Comment by atuchin-m** [comment]
The fix: https://github.com/brave/brave-core/pull/14281

**Pull request merged**